mos_tests/neutron/python_tests/test_restarts.py

```diff
@@ -71,21 +74,37 @@ def prepare_openstack(self, setup):
         self.check_vm_connectivity()
 
         # Find a primary contrloller
-        for node in self.env.get_all_nodes():
-            # TBD this is a temp solution.
-            # Te be replaced by 'heira role' in next commit
-            if node.data['name'] == 'slave-01_controller':
-                self.primary_node = DevopsClient.get_devops_node(
-                                        node.data['name'].split('_')[0])
-                self.primary_host = node.data['fqdn']
-                break
+        self.primary_node = DevopsClient.get_devops_node(
+            self.env.primary_controller.data['name'].split('_')[0])
```

Comment:
If you deploy OpenStack by hand - name of nodes inside fuel may be any string. I believe it will better to determine devops node by MAC address.